### PR TITLE
feat(path): add xcp_path_parent_dir method

### PR DESCRIPTION
**Pull request by Wescoeur**

### CODE REVIEW

File: src/path.c [hunk 1]
```diff
@@ -50,3 +51,70 @@ char *xcp_path_combine (const char *pathname, const char *subpath) {
 
   return res;
 }
+
+static inline const char *get_first_slash (const char *start, const char *pos) {
```
Comment (benjamreis) [inline src/path.c:55]: You run the loop backward, from `pos - 1` to `start`, so doesn't it return the last slash position?

File: src/path.c [hunk 2]
```diff
@@ -50,3 +51,70 @@ char *xcp_path_combine (const char *pathname, const char *subpath) {
...
+// "/gfrg////grg" => /gfrg
+// "titia" => .
+// "/toto/" => /
+// "//" => //
+// "///" => /
+// "////" => /
+// "//a" => //
+// "///b" => /
+// "////c" => /
+// "////c/d" => ////c
```
Comment (benjamreis) [inline src/path.c:80]: That's exhaustive!

File: src/path.c (hunk 1 above)
Comment (Wescoeur) [reply src/path.c:55]: The loop try to find the first slash before a given position of consecutive slashes until a non-slash char has been found.

Comment (benjamreis) [reply src/path.c:55]: ok good to merge

**Pull request merged**